yices2/src/yices2_factory.cpp

```diff
@@ -0,0 +1,25 @@
+#include "yices2_factory.h"
+
+#include "yices2_solver.h"
+
+namespace smt {
+
+bool initialized = false;
+
+/* Yices2SolverFactory implementation */
+SmtSolver Yices2SolverFactory::create()
+{
+  // Must initialize only once.
+  // Different instances of the solver will have
+  // different contexts.
+  if (!initialized)
+  {
+    yices_init();
+    initialized = true;
+  }
+
+  return std::make_unique<Yices2Solver>();
```

Comment:
Travis failed again, sorry :/ I changed SmtSolver to be a `shared_ptr`.
```suggestion
  return std::make_shared<Yices2Solver>();
```